toolkit/tools/pkg/imagecustomizerlib/imageutils.go

```diff
@@ -0,0 +1,195 @@
+	}
+
+	return imageConnection, nil
+
+}
+
+func connectToExistingImageHelper(imageConnection *ImageConnection, imageFilePath string,
+	buildDir string, chrootDirName string,
+) error {
+	// Connect to image file using loopback device.
+	err := imageConnection.ConnectLoopback(imageFilePath)
+	if err != nil {
+		return err
+	}
+
+	// Look for all the partitions on the image.
+	newMountDirectories, mountPoints, err := findPartitions(buildDir, imageConnection.Loopback().DevicePath())
+	if err != nil {
+		return fmt.Errorf("failed to find disk partitions:\n%w", err)
+	}
+
+	// Create chroot environment.
+	imageChrootDir := filepath.Join(buildDir, chrootDirName)
+
+	err = imageConnection.ConnectChroot(imageChrootDir, false, newMountDirectories, mountPoints)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createNewImage(filename string, diskConfig imagecustomizerapi.Disk,
+	partitionSettings []imagecustomizerapi.PartitionSetting, bootType imagecustomizerapi.BootType, buildDir string,
+	chrootDirName string, installOS installOSFunc,
+) (*ImageConnection, error) {
+	imageConnection := &ImageConnection{}
+
+	err := createNewImageHelper(imageConnection, filename, diskConfig, partitionSettings, bootType, buildDir,
+		chrootDirName, installOS,
+	)
+	if err != nil {
+		imageConnection.Close()
+		return nil, fmt.Errorf("failed to create new image:\n%w", err)
+	}
+
+	return imageConnection, nil
+}
+
+func createNewImageHelper(imageConnection *ImageConnection, filename string, diskConfig imagecustomizerapi.Disk,
+	partitionSettings []imagecustomizerapi.PartitionSetting, bootType imagecustomizerapi.BootType, buildDir string,
+	chrootDirName string, installOS installOSFunc,
+) error {
+	// Convert config to image config types, so that the imager's utils can be used.
+	imagerBootType, err := bootTypeToImager(bootType)
+	if err != nil {
+		return err
+	}
+
+	imagerDiskConfig, err := diskConfigToImager(diskConfig)
+	if err != nil {
+		return err
+	}
+
+	imagerPartitionSettings, err := partitionSettingsToImager(partitionSettings)
+	if err != nil {
+		return err
+	}
+
+	// Sort the partitions so that they are mounted in the correct oder.
+	sort.Slice(imagerPartitionSettings, func(i, j int) bool {
+		return imagerPartitionSettings[i].MountPoint < imagerPartitionSettings[j].MountPoint
+	})
+
+	// Create imager boilerplate.
+	mountPointMap, err := createImageBoilerplate(imageConnection, filename, buildDir, chrootDirName, imagerDiskConfig,
+		imagerPartitionSettings)
+	if err != nil {
+		return err
+	}
+
+	// Install the OS.
+	err = installOS(imageConnection.Chroot())
+	if err != nil {
+		return err
+	}
+
+	// Configure the boot loader.
+	err = installutils.ConfigureDiskBootloader(imagerBootType, false, false, imagerPartitionSettings,
+		configuration.KernelCommandLine{}, imageConnection.Chroot(), imageConnection.Loopback().DevicePath(),
+		mountPointMap, diskutils.EncryptedRootDevice{}, diskutils.VerityDevice{}, false /*enableGrubMkconfig*/)
```

Comment:
@camelron @cwize1
Added the `faslse /*enabledGrubMkconfig*/` here.

Reply:
Okay, I made a rough translation of EnableGrubMkconfig into the imagecustomizer. See the commit here: https://github.com/microsoft/CBL-Mariner/commit/b652b61872de1dfa6e1310f9d65e51b87242b8fc 

@cwize1 I think we can do this a little cleaner; why not pass down the entire Config struct to createNewImage instead of its separate components (diskConfig, etc)? It seems like we hard-code  diskConfig to the first disk in the Config Disks[] anyways, but I feel I'm missing some context?

Reply:
I think it is fine to leave it as `false` for now.

The change will probably need to be slightly more complex than the one Camelron provided, since specifying `EnableGrubMkconfig` probably isn't valid when the partitions aren't customized.